resources/default.rb

```diff
@@ -4,19 +4,69 @@
 property :artifactoryonline, :kind_of => String
 property :repository, :kind_of => String, :required => true
 property :repository_path, :kind_of => String, :required => true
+property :artifact_name, :kind_of => String
 property :artifactory_username, :kind_of => String
 property :artifactory_password, :kind_of => String
+property :fetch_highest, :kind_of => [TrueClass, FalseClass]
 
 property :group, :kind_of => [Integer, String]
 property :mode, :kind_of => [Integer, String]
 property :owner, :kind_of => [Integer, String]
 
 action_class do
   require "uri"
+  require 'net/http'
+  require 'json'
 
   include ::ArtifactoryArtifact::Helper
 
+  def fetching_highest(new_resource)
+
+    if new_resource.artifact_name.nil?
+      raise Chef::Exceptions::ValidationFailed, "artifact_name is a required property, when fetch_highest is set true"
+    end
+
+    # Fetching list of artifacts from artifactory
+    url = URI("#{new_resource.artifactory_url}/api/search/aql/")
+    http = Net::HTTP.new(url.host, url.port)
+    request = Net::HTTP::Post.new(url)
+
+    request['Content-Type'] = 'text/plain'
+    request['Accept'] = '*/*'
+    request['Host'] = url.host
+    request['Connection'] = 'keep-alive'
+    request.basic_auth new_resource.artifactory_username, new_resource.artifactory_password
+    request.body = "items.find({\"repo\":{\"$eq\":\"   #{new_resource.repository}\"}})"
+    response = http.request(request)
+    
+    unless response.kind_of? Net::HTTPSuccess
+      Chef::Log.error(response.body)
+      raise Chef::Exceptions::InvalidSearchQuery , "Artifactory Search Failed"
+    end
+
+    parsed_json = JSON.parse(response.body)
+    name_re = new_resource.artifact_name.gsub(/HIGHEST/, '([\.0-9]+)')
+
+    # Adding Version tag to parsed_json
+    parsed_json['results'].each do |res|
+      output = res['name']
+      output =~ /^#{name_re}$/
+      res['version'] = $1
+    end
+
+    #Sorting versions and picking highest value
+    parsed_json['results'].sort! { |x,y|
+      Chef::Provider::Package::Yum::RPMUtils.rpmvercmp(x['version'], y['version'])
```

Comment:
It seems like RPMUtils.rpmvercmp is actually a Ruby impl and is NOT relying on any RedHat executables. Just in case, please consider adding explicit require line for the library class.